src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -334,7 +336,15 @@ private static async System.Threading.Tasks.Task SaveLogFileAsync(string filePat
             }
         }
 
-        internal static async System.Threading.Tasks.Task ProcessSarifLogAsync(SarifLog sarifLog, string logFilePath, bool showMessageOnNoResults, bool cleanErrors, bool openInEditor)
+        internal static async Task ProcessSarifLogAsync(Stream stream, bool showMessageOnNoResults, bool cleanErrors, bool openInEditor)
+        {
+            SarifLog sarifLog = SarifLog.Load(stream);
+            string logFilePath = Path.GetTempFileName();
```

Comment:
Interesting... I'd like to understand the ramifications of this.

Reply:
vs allowing it to be null

---
In reply to: [519462499](https://github.com/microsoft/sarif-visualstudio-extension/pull/259#discussion_r519462499) [](ancestors = 519462499)

Reply:
Hmm. In the PR description I mentioned a choice between a "real" file and a random (GUID) string, but I didn't consider leaving it blank. Let's try that...

> Since the file isn't really used, I considered just using GUID string as the file path, without actually creating the empty file. I chose to create the file since if in future, we introduce gestures that actually make use of the file (for example, writing a modified set of results back to it), it will be good to have it around. Otherwise, we'd be writing GUID-named files to whatever the current directory is.

---
In reply to: [519462520](https://github.com/microsoft/sarif-visualstudio-extension/pull/259#discussion_r519462520) [](ancestors = 519462520,519462499)

Reply:
You are right. It works just fine without a log file path name.

---
In reply to: [519464163](https://github.com/microsoft/sarif-visualstudio-extension/pull/259#discussion_r519464163) [](ancestors = 519464163,519462520,519462499)